components/fonts/shapers/harfrust.rs

```diff
@@ -0,0 +1,159 @@
+use app_units::Au;
+use euclid::default::Point2D;
+use harfrust::{
+    Feature, FontRef as HarfRustFontRef, GlyphBuffer, Script, ShaperData, Tag, UnicodeBuffer,
+};
+use num_traits::Zero as _;
+use smallvec::SmallVec;
+
+use super::{ShapedGlyphEntry, THarfShapedGlyphData, THarfShaper, unicode_to_hb_script};
+use crate::{Font, FontData, ShapingFlags, fixed_to_float};
+
+#[inline(always)]
+fn au_from_fixed_glyph_info(i: i32, ppem: f64) -> Au {
+    Au::from_f64_px(fixed_to_float(16, i) * ppem * 64.0)
```

Comment:
HarfRust always shapes in font units so we don’t have a fixed point value here. I believe the proper code is `i as f64 * scale` where `scale` is `ppem / upem`. You can grab the `upem` value with `Shaper::units_per_em()` when you’re shaping a run and precompute the scale there.

Reply:
This seems to have fixed things - thanks!